packages/amplify-e2e-tests/src/__tests__/function-permissions.test.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  addNotificationChannel,
+  addFunction,
+  amplifyPushUpdate,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+} from '@aws-amplify/amplify-e2e-core';
+import { getShortId } from '../import-helpers';
+
+describe('amplify add function with permissions', () => {
+  let projRoot: string;
+
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('function-permissions');
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
+  });
+
+  it('add notifications and function permissions', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+
+    const settings = { resourceName: `notifications${getShortId()}` };
+    await addNotificationChannel(projRoot, settings, 'SMS');
+    await addFunction(
+      projRoot,
+      {
+        name: 'testFunction',
+        functionTemplate: 'Hello World',
+        additionalPermissions: {
+          permissions: ['notifications'],
+          resources: [settings.resourceName],
+          choices: ['auth', 'analytics', 'notifications'],
+          operations: ['create', 'read'],
+        },
+      },
+      'nodejs',
+    );
+    await amplifyPushUpdate(projRoot);
```

Comment:
Is there anything permissions related we can verify after `push`?

Reply:
not really, notifications doesn't actually support permissions, we print this warning: `notifications category does not support resource policies yet`, the issue was that previously the error was propagated up preventing the customer to continue